.github/workflows/lint.yaml

```diff
@@ -0,0 +1,43 @@
+name: "lint"
+
+on:
+  pull_request:
+    types: ["opened", "reopened"]
+  push:
+  schedule:
+    # Run at midnight UTC every day with 15 minutes delay added to avoid high load periods
+    - cron: "15 0 * * *"
+  workflow_dispatch:
+
+permissions:
```

Comment:
Would it be better to move the `checks` and `contents` permissions under the job? 

I was reading this and realize the permissions can be controlled per job: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idpermissions

Reply:
Sure